### PR TITLE
Basic program with one example each of FTP, SFTP and API upload

### PR DESCRIPTION
**Pull request by rhigman**
Further work to be done on this, but feedback would be useful at this stage.

### CODE REVIEW

File: uploader.py [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+
+    def get_cover_image(self):
+        """
+        Retrieve work cover image from URL specified in work metadata
+        (required by some platforms)
+        """
+        # Extract cover URL from Thoth metadata
+        cover_url = self.get_cover_url()
+
+        match cover_url.split('.')[-1]:
```
Comment (ja573) [inline uploader.py:62]: could be worth matching on the lowercased URL

File: souploader.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        new_dir = date.today().isoformat()
+
+        # Metadata file format TBD: use CSV for now
+        metadata_bytes = self.get_formatted_metadata('csv::thoth')
+        cover_bytes = self.get_cover_image()
+        pdf_bytes = self.get_pdf_bytes()
+
+        files = [
+            ('{}.csv'.format(filename), metadata_bytes),
+            ('{}.jpg'.format(filename), cover_bytes),
```
Comment (ja573) [inline souploader.py:47]: how would this work for png images?

File: iauploader.py [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        logging.info(
+            'Successfully uploaded to Internet Archive at archive.org/details/{}'.format(filename))
+
+    def parse_metadata(self):
+        """Convert work metadata into Internet Archive format"""
+        work_metadata = self.metadata.get('data').get('work')
+        ia_metadata = {
+            'title': work_metadata.get('title'),
+            'publisher': self.get_publisher_name(),
+            'creator': work_metadata.get('copyrightHolder'),
```
Comment (ja573) [inline iauploader.py:55]: I'm not entirely sure the creator should be the copyright holder. Could this be perhaps a concatenation of the main contributions?

**Review by ja573 — APPROVED** [review]
That's all looking good.

Down the line it'd be good to place the uploaders into their own library, which disseminator could then import. That way we can potentially handle multiple uploads from the disseminator at once

File: requirements.txt [hunk 4]
```diff
@@ -0,0 +1,5 @@
+internetarchive
```
[the remark this answers is not in the document]
Comment (rhigman) [reply requirements.txt:1]: Agreed. I'll need to pin `thothlibrary` to a version that contains https://github.com/thoth-pub/thoth-client/pull/15, though - is there likely to be another release soon?

File: souploader.py (hunk 2 above)
Comment (rhigman) [reply souploader.py:47]: Oops, good spot, thanks.

File: requirements.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (ja573) [reply requirements.txt:1]: That should now be included in thothlibrary==0.14.0

File: iauploader.py (hunk 3 above)
Comment (rhigman) [reply iauploader.py]: Agreed - this was intended as a placeholder, although I've now got a handle on repeatable fields (e.g. creator) and added some more. As discussed, we may well move this logic to the export server anyway (creating a new JSON output).

**Pull request merged**